src/protocol.cpp

```diff
@@ -226,3 +226,27 @@ const std::vector<std::string>& getAllNetMessageTypes()
 {
     return allNetMessageTypesVec;
 }
+
+const std::vector<std::string>& getTierTwoNetMessageTypes()
+{
+    // Note: keep this updated with allNetMessageTypes
+    const static std::vector<std::string> allTierTwoMessageTypesVec {
+            NetMsgType::SPORK,
+            NetMsgType::MNWINNER,
+            "mnodescanerr",
+            NetMsgType::BUDGETVOTE,
+            NetMsgType::BUDGETPROPOSAL,
+            NetMsgType::FINALBUDGET,
+            NetMsgType::FINALBUDGETVOTE,
+            "mnq",
+            NetMsgType::MNBROADCAST,
+            NetMsgType::MNPING,
+            "dstx",  // deprecated
+            NetMsgType::GETMNWINNERS,
+            NetMsgType::GETMNLIST,
+            NetMsgType::BUDGETVOTESYNC,
+            NetMsgType::GETSPORKS,
+            NetMsgType::SYNCSTATUSCOUNT
+    };
+    return allTierTwoMessageTypesVec;
+}
```

Comment:
I'm not a fan of this kind of duplication, where two static constant lists depend on each other, and, when we modify one we have to remember to update the other one too.
It's easy to miss the comments and introduce bugs this way (it happened to me in the past).

I propose this alternative:
```patch
Index: src/protocol.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/protocol.cpp b/src/protocol.cpp
--- a/src/protocol.cpp	(revision f16f4224515ad0468c54795ccf055e8743c4858d)
+++ b/src/protocol.cpp	(date 1626951151800)
@@ -80,7 +80,7 @@
     "filtered block", // Should never occur
     "ix",   // deprecated
     "txlvote", // deprecated
-    NetMsgType::SPORK,
+    NetMsgType::SPORK,           // --- tiertwoNetMessageTypes start here ---
     NetMsgType::MNWINNER,
     "mnodescanerr",
     NetMsgType::BUDGETVOTE,
@@ -98,6 +98,7 @@
     NetMsgType::SYNCSTATUSCOUNT
 };
 const static std::vector<std::string> allNetMessageTypesVec(allNetMessageTypes, allNetMessageTypes + ARRAYLEN(allNetMessageTypes));
+const static std::vector<std::string> tiertwoNetMessageTypesVec(std::find(allNetMessageTypesVec.begin(), allNetMessageTypesVec.end(), NetMsgType::SPORK), allNetMessageTypesVec.end());
 
 CMessageHeader::CMessageHeader(const MessageStartChars& pchMessageStartIn)
 {
@@ -229,24 +230,5 @@
 
 const std::vector<std::string>& getTierTwoNetMessageTypes()
 {
-    // Note: keep this updated with allNetMessageTypes
-    const static std::vector<std::string> allTierTwoMessageTypesVec {
-            NetMsgType::SPORK,
-            NetMsgType::MNWINNER,
-            "mnodescanerr",
-            NetMsgType::BUDGETVOTE,
-            NetMsgType::BUDGETPROPOSAL,
-            NetMsgType::FINALBUDGET,
-            NetMsgType::FINALBUDGETVOTE,
-            "mnq",
-            NetMsgType::MNBROADCAST,
-            NetMsgType::MNPING,
-            "dstx",  // deprecated
-            NetMsgType::GETMNWINNERS,
-            NetMsgType::GETMNLIST,
-            NetMsgType::BUDGETVOTESYNC,
-            NetMsgType::GETSPORKS,
-            NetMsgType::SYNCSTATUSCOUNT
-    };
-    return allTierTwoMessageTypesVec;
-}
\ No newline at end of file
+    return tiertwoNetMessageTypesVec;
+}
```

Reply:
sounds good, it will still need a good comment to modify the vector in case of adding non tier two related messages at the end of the vector.